### PR TITLE
ONEM-28398 fix issues after uninstalling app with uninstallType=upgrade

### PR DESCRIPTION
**Pull request by sverkoye**
After uninstalling an app with uninstallType=upgrade:

* getList() should still return type+id but without any installed apps (=versions)
* getStorageDetails() should return a storagePath and allow calling with empty version
* fix related problem in doMaintenance()
* allow uninstall (type=full) to be called with empty version so that the remaing storage dir and app record can get removed
* make sure metadata is automatically removed when removing an app

### CODE REVIEW

**Review by piotr-marcinkowski-red — APPROVED** [review]

File: LISA/Executor.cpp [hunk 1]
```diff
@@ -216,7 +216,12 @@ uint32_t Executor::Uninstall(const std::string& type,
         return ERROR_WRONG_PARAMS;
     }
 
-    if (! isAppInstalled(type, id, version)) {
+    if (version.empty() && !type.empty() && !id.empty() && uninstallType == "full") {
```
Comment (piotr-marcinkowski-red) [inline LISA/Executor.cpp:219]: What's the actual scenario here? How an app can fall into this case while uninstalling it?

File: LISA/Executor.cpp [hunk 2]
```diff
@@ -343,39 +348,31 @@ uint32_t Executor::GetProgress(const std::string& handle, std::uint32_t& progres
...
                            Filesystem::StorageDetails& details)
 {
     namespace fs = Filesystem;
-    if(!getStorageParamsValid(type, id, version)) {
-        return ERROR_WRONG_PARAMS;
-    }
+
     try {
-        if(type.empty()) {
+        if(type.empty() && id.empty() && version.empty()) {
```
Comment (piotr-marcinkowski-red) [inline LISA/Executor.cpp:359]: This got more complex than it was before with just type.empty() check. Could you add a comment what is happening here?

File: LISA/Executor.cpp (hunk 1 above)
Comment (sverkoye) [reply LISA/Executor.cpp:224]: Added extra comments

File: LISA/Executor.cpp (hunk 2 above)
Comment (sverkoye) [reply LISA/Executor.cpp:370]: Added extra comments

**Review by piotr-marcinkowski-red — APPROVED** [review]

**Pull request merged**